mozjs/js/rust/src/magicdom/domquad.rs

```diff
@@ -20,6 +25,22 @@ magic_dom! {
     }
 }
 
+impl NumSlots for DOMQuad {
+    const NUM_SLOTS: u32 = <dompoint::DOMPoint as NumSlots>::NUM_SLOTS_PTR * 4;
+}
+
+impl ToFromJsSlots for DOMQuad {
```

Comment:
Both the `NumSlots` and `ToFromJsSlots` impls for magic structs should be emitted by `magic_codegen` since they will always be stored as `ObjectValue` and then can be read out into `Option<MagicStructType>` with something like:

```rust
if slot_value.is_object() {
    MagicDomStructType::from_object(slot_value.to_object())
} else {
    None
}
```